core/configuration/src/main/kotlin/guildsettings/GuildSettingsDatabaseFactory.kt

```diff
@@ -0,0 +1,15 @@
+package guildsettings
+
+import sqldelight.SQLDelightDrivers
+
+object GuildSettingsDatabaseFactory {
+    private val instances: MutableMap<String, GuildSettingsDatabase> = mutableMapOf()
+
+    fun instance(name: String): GuildSettingsDatabase {
+        return instances[name] ?: synchronized(this) {
```

Comment:
synchronizing function is later called within coroutine. This is not recommended. Recommendation: Make function suspending or return a Deferred for the database.

Reply:
this isn't a suspend function. It's perfectly fine to use synchronization here, in fact this is the only way to have a secure singleton lazy initialization (it's called Double Checked Locking singleton.

If you had a suspend function here with a caching side effect like that you would still have to use a `Mutex` (which is basically the synchronize in the coroutine world) but I don't think we'll ever want a suspend function to initialize our modules

Reply:
Closing so we have one conversation going on for this